server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/InheritanceAction.java

```diff
@@ -70,26 +66,21 @@ public void define(NewController context) {
       .setHandler(this)
       .setResponseExample(getClass().getResource("example-inheritance.json"));
 
-    QProfileIdentificationParamUtils.defineProfileParams(inheritance, languages);
+    QProfileRef.defineParams(inheritance, languages);
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    DbSession session = dbClient.openSession(false);
+    DbSession dbSession = dbClient.openSession(false);
     try {
-      String profileKey = QProfileIdentificationParamUtils.getProfileKeyFromParameters(request, profileFactory, session);
-      QualityProfileDto profile = dbClient.qualityProfileDao().selectByKey(session, profileKey);
-      if (profile == null) {
-        throw new NotFoundException(String.format("Could not find a quality profile with key %s", profileKey));
-      }
-
-      List<QProfile> ancestors = profileLookup.ancestors(profile, session);
-      List<QualityProfileDto> children = dbClient.qualityProfileDao().selectChildren(session, profileKey);
+      QualityProfileDto profile = profileFactory.find(QProfileRef.from(request), dbSession);
```

Comment:
the problem is I'm so used to have DbSession as the first parameter that I misread your code. The problem is between the chair and the screen, and it's not on your side! : )